Lib/threading.py

```diff
@@ -1475,11 +1482,34 @@ def __init__(self):
         with _active_limbo_lock:
             _active[self._ident] = self
 
+# Helper thread-local instance to detect when a _DummyThread
+# is collected. Not a part of the public API.
+_thread_local_info = local()
+
+class __DeleteDummyThreadOnDel:
```

Comment:
Any specific reason to make it private and not just protected?

Reply:
Well, it's not part of the public API and no user should use it. I'd make `_thread_local_info` private too (but then the name mangling make it hard for other places to access it, so, I just went with protected).

If it's a problem I can change to make it protected though.